tf_agents/agents/ppo/ppo_agent.py

```diff
@@ -487,11 +492,13 @@ def _train(self, experience, weights):
         experience.observation, experience.step_type, policy_state=policy_state)
     value_preds = tf.stop_gradient(value_preds)
 
-    valid_mask = ppo_utils.make_timestep_mask(next_time_steps)
```

Comment:
pass `train_only_on_partial_episodes` to `make_timestep_mask` and build an appropriate mask.

keep `valid_mask` in the rest of code.

Reply:
Used `train_on_partial_episodes` instead of `train_only_on_partial_episodes` as it should also work in episodic envs as well.

Bootstrapped the reward using the value function as well.